spring/spring4/spring4-common/src/main/java/org/infinispan/spring/provider/CacheDelegate.java

```diff
@@ -16,6 +20,8 @@
  */
 class CacheDelegate implements Cache {
 
+   private final Map<Object, StampedLock> synchronousGetLocks = new HashMap<>();
```

Comment:
I might be missing something obvious, but this should be a *concurrent* map, shouldn't it?